CMakeLists.txt

```diff
@@ -90,15 +91,25 @@ if(TARGET Boost::program_options)
     target_link_libraries(sphinxsys_core INTERFACE Boost::program_options)
 endif()
 
+if(SPHINXSYS_USE_SYCL)
+    if(NOT SPHINXSYS_SYCL_TARGETS)
+        set(SPHINXSYS_SYCL_TARGETS nvptx64-nvidia-cuda)
```

Comment:
As far as I know spir64 is based on OpenCL, and not all systems might come with it installed (mine doesn't for example), so that could also not be considered as a safe default case.

It might be better not to have a default value in the first place and require the user to pass a `SPHINXSYS_SYCL_TARGETS` alongside `SPHINXSYS_USE_SYCL`

Another thing to reason about is the sycl selector to use: for this PR I used `gpu_selector_v`, but to keep it general enough `default_selector_v` may be preferred. As an additional feature we could give the user (of the library or the final user) the possibility to set their own selector.

Reply:
spir64 seems to be the default when no target is given anyway. 

> spir64    Tells the compiler to use default heuristics for SPIR64-based devices. This is the default. You can also specify this value as spir64-unknown-unknown.

Leaving it empty by default is best because user must be forced to pick. If several target can be input, the CMake combo box might not be sufficient